pkg/stats/stats.go

```diff
@@ -36,149 +45,169 @@ const (
 
-	// TotalAssignedIDsUpdated represents the number of AzureAssignedIdentities updated in a given sync cycle.
-	TotalAssignedIDsUpdated Type = "Number of assigned ids updated in this sync cycle"
+	// TotalAzureAssignedIdentitiesDeleted represents the number of AzureAssignedIdentities deleted in a given sync cycle.
+	TotalAzureAssignedIdentitiesDeleted Type = "Number of AzureAssignedIdentities deleted in this sync cycle"
 
-	// TotalAssignedIDsDeleted represents the number of AzureAssignedIdentities deleted in a given sync cycle.
-	TotalAssignedIDsDeleted Type = "Number of assigned ids deleted in this sync cycle"
+	// FindAzureAssignedIdentitiesToDelete represents the duration it takes to generate a list of AzureAssignedIdentities to be deleted.
+	FindAzureAssignedIdentitiesToDelete Type = "Find AzureAssignedIdentities to delete"
 
-	// FindAssignedIDDel represents the duration it takes to generate a list of AzureAssignedIdentities to be deleted.
-	FindAssignedIDDel Type = "Find assigned ids to delete"
+	// FindAzureAssignedIdentitiesToCreate represents the duration it takes to generate a list of AzureAssignedIdentities to be created.
+	FindAzureAssignedIdentitiesToCreate Type = "Find AzureAssignedIdentities to create"
 
-	// FindAssignedIDCreate represents the duration it takes to generate a list of AzureAssignedIdentities to be created.
-	FindAssignedIDCreate Type = "Find assigned ids to create"
+	// DeleteAzureAssignedIdentity represents the duration it takes to delete an AzureAssignedIdentity.
+	DeleteAzureAssignedIdentity Type = "AzureAssignedIdentity deletion"
 
-	// AssignedIDDel represents the duration it takes to delete an AzureAssignedIdentity.
-	AssignedIDDel Type = "Assigned ID deletion"
+	// CreateAzureAssignedIdentiy represents the duration it takes to create an AzureAssignedIdentity.
+	CreateAzureAssignedIdentiy Type = "AzureAssignedIdentity creation"
 
-	// AssignedIDAdd represents the duration it takes to create an AzureAssignedIdentity.
-	AssignedIDAdd Type = "Assigned ID addition"
+	// UpdateAzureAssignedIdentity represents the duration it takes to update an AzureAssignedIdentity.
+	UpdateAzureAssignedIdentity Type = "AzureAssignedIdentity update"
 
-	// TotalCreateOrUpdate represents the duration it takes to create or update a given list of AzureAssignedIdentities.
-	TotalCreateOrUpdate Type = "Total time to assign or update IDs"
+	// TotalAzureAssignedIdentitiesCreateOrUpdate represents the duration it takes to create or update a given list of AzureAssignedIdentities.
+	TotalAzureAssignedIdentitiesCreateOrUpdate Type = "Total time to assign or update AzureAssignedIdentities"
 )
 
-// Init initializes the maps uesd to store the stats.
+// Init initializes the maps uesd to store the
 func Init() {
-	GlobalStats = make(map[Type]time.Duration)
-	CountStats = make(map[Type]int)
+	globalStats = make(map[Type]time.Duration)
+	minMaxTimeStats = make(map[Type]minMaxTime)
+	countStats = make(map[Type]int)
 	mutex = &sync.RWMutex{}
```

Comment:
If we are not using read locks, we should just change this to `sync.Mutex`

Reply:
We used it in `Print()` method.